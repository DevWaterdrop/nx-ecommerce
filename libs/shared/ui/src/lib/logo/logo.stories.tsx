import { Story, Meta } from '@storybook/react';
import { Logo, LogoProps } from './logo';

export default {
  component: Logo,
  title: 'Components/Logo',
} as Meta;

const Template: Story<LogoProps> = (args) => <Logo {...args} />;

export const Primary = Template.bind({});
Primary.args = {};
