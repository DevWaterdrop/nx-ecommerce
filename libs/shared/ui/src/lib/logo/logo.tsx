/* eslint-disable-next-line */
export interface LogoProps {}

export const Logo: React.FC<LogoProps> = (props) => {
  const [firstWord, secondWord] = 'fake shop'.split(' ');

  return (
    <a
      className="flex flex-col -mb-4 w-max items-center"
      href="/"
      title="fake shop"
    >
      <div
        className="flex font-bold animate-jello text-2xl leading-4 uppercase"
        aria-hidden="true"
        data-testid="first-word"
      >
        {firstWord.split('').map((char, index) => (
          <span
            className={'inline-block animate-bounce'}
            style={{ animationDelay: `-${index * 50}ms` }}
            key={char}
          >
            {char}
          </span>
        ))}
      </div>
      <div
        className="flex font-bold animate-jello text-2xl uppercase"
        aria-hidden="true"
        data-testid="second-word"
      >
        {secondWord.split('').map((char, index) => (
          <span
            className={'inline-block animate-bounce'}
            style={{ animationDelay: `-${index * 50}ms` }}
            key={char}
          >
            {char}
          </span>
        ))}
      </div>
    </a>
  );
};
