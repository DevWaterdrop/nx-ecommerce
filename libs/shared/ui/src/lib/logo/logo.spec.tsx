import { render, RenderResult, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';

import { Logo } from './logo';

describe('Logo', () => {
  let renderResult: RenderResult;

  beforeEach(() => {
    renderResult = render(<Logo />);
  });

  it('should render successfully', () => {
    const { baseElement } = renderResult;

    const tree = renderer.create(<Logo />).toJSON();
    expect(tree).toMatchSnapshot();

    expect(baseElement).toBeTruthy();
  });

  it('contain all chars in two words', () => {
    const [firstWord, secondWord] = 'fake shop'.split(' ');

    const firsWordContainerElement = screen.getByTestId('first-word');

    firstWord.split('').forEach((char) => {
      const charElement = screen.getByText(char);
      expect(charElement).toBeInTheDocument();
      expect(charElement.parentElement).toBe(firsWordContainerElement);
    });

    const secondWordContainerElement = screen.getByTestId('second-word');
    secondWord.split('').forEach((char) => {
      const charElement = screen.getByText(char);
      expect(charElement).toBeInTheDocument();
      expect(charElement.parentElement).toBe(secondWordContainerElement);
    });
  });
});
