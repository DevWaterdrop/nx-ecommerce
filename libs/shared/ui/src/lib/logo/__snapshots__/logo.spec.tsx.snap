// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo should render successfully 1`] = `
<a
  className="-mb-4 w-max flex flex-col items-center"
  href="/"
  onClick={[Function]}
  onMouseEnter={[Function]}
  title="fake shop"
>
  <div
    aria-hidden="true"
    className="flex font-bold text-2xl leading-4 uppercase pointer-events-none select-none animate-jello"
    data-testid="first-word"
  >
    <span
      className="inline-block animate-bounce"
      style={
        Object {
          "animationDelay": "-0ms",
        }
      }
    >
      f
    </span>
    <span
      className="inline-block animate-bounce"
      style={
        Object {
          "animationDelay": "-50ms",
        }
      }
    >
      a
    </span>
    <span
      className="inline-block animate-bounce"
      style={
        Object {
          "animationDelay": "-100ms",
        }
      }
    >
      k
    </span>
    <span
      className="inline-block animate-bounce"
      style={
        Object {
          "animationDelay": "-150ms",
        }
      }
    >
      e
    </span>
  </div>
  <div
    aria-hidden="true"
    className="flex font-bold text-2xl uppercase pointer-events-none select-none animate-jello"
    data-testid="second-word"
  >
    <span
      className="inline-block animate-bounce"
      style={
        Object {
          "animationDelay": "-0ms",
        }
      }
    >
      s
    </span>
    <span
      className="inline-block animate-bounce"
      style={
        Object {
          "animationDelay": "-50ms",
        }
      }
    >
      h
    </span>
    <span
      className="inline-block animate-bounce"
      style={
        Object {
          "animationDelay": "-100ms",
        }
      }
    >
      o
    </span>
    <span
      className="inline-block animate-bounce"
      style={
        Object {
          "animationDelay": "-150ms",
        }
      }
    >
      p
    </span>
  </div>
</a>
`;
